crates/btc-relay/src/lib.rs

```diff
@@ -1034,7 +1034,7 @@ impl<T: Config> Pallet<T> {
     /// * `block_height` - current block height
     fn get_last_retarget_time(chain_id: u32, block_height: u32) -> Result<u64, DispatchError> {
         let block_chain = Self::get_block_chain_from_id(chain_id)?;
-        let period_start_height = block_height - block_height % DIFFICULTY_ADJUSTMENT_INTERVAL;
```

Comment:
Note: this is always zero since we only enter this function if `block_height % DIFFICULTY_ADJUSTMENT_INTERVAL == 0`

Reply:
I mean, that's arguably correct. You could argue that the most recent retarget will have been this block. Then the way to fix it would be this:

```.diff
fn compute_new_target(
        prev_block_header: &RichBlockHeader<T::BlockNumber>,
        block_height: u32,
    ) -> Result<U256, DispatchError> {
        // time of last retarget (first block in current difficulty period)
-        let first_block_time = Self::get_last_retarget_time(prev_block_header.chain_id, block_height)?;
+        let first_block_time = Self::get_last_retarget_time(prev_block_header.chain_id, prev_block_header.block_height)?;
```

Either change would work

Reply:
right, that would also work. I find my suggestion a tad more intuitive though. @gregdhill any opinion?

Reply:
Both would work but let's just use your change @nud3l. Can you use saturated math though?